### PR TITLE
Remove debian packages from repo

### PR DESCRIPTION
**Pull request by d-513**

### CODE REVIEW

**Comment by d-513** [comment]
I also think we should move packaging out of this repo asap

**Comment by redstrate** [comment]
> I also think we should move packaging out of this repo asap

+1, it makes no sense to include it in this repository, make people clone the whole thing just for a .spec file that downloads another tarball from Github anyway

**Comment by d-513** [comment]
Yes and also it’s confusing as to what version is used

File: packages/debian/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        "git+https://github.com/MultiMC/libnbtplusplus"
+        "git+https://github.com/PolyMC/quazip")
+
+sha256sums=('SKIP'
+            'SKIP'
+            'SKIP')
+
+prepare() {
+  cd "${srcdir}/PolyMC"
+  git checkout ${pkgver}
```
Comment (hwittenborn) [inline packages/debian/PKGBUILD]: You should be able to replace this with the following under your `source` array:

```sh
source=("git+https://github.com/PolyMC/PolyMC#tag=${pkgver}")
```

**Review by hwittenborn — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
I've outlined my reasons for leaving packaging in the repo. Nix absolutely requires to be in the same repo to my knowledge, and it's very convenient to be able to build the Deb straight in the repository. Especially for Jenkins, as it means a full rebuild is not necessary for the deb.

**Comment by d-513** [comment]
> 

So I should close this PR?

**Comment by ghost** [comment]
No, I believe we should leave packaging here. If makedeb works and builds everything properly then this should be good to merge

**Comment by LennyMcLennington** [comment]
> Nix absolutely requires to be in the same repo to my knowledge

Hard to believe that every single program that nix supports has the packaging scripts in their repo. Why is it absolutely required specifically for this program?

> Especially for Jenkins, as it means a full rebuild is not necessary for the deb.

We can set up Jenkins to have a separate packaging workflow that triggers after a successful build and uses the artifact produced from that to package it.

**Comment by d-513** [comment]
since we are moving to MPR and AUR this is not problem anymore  and we could move the packaging to a different repo.
In that repo we should have scripts like update.sh which jenkins will run to upload packages to mpr, aur and copr
No build neccesary except the normal system, portable, windows, appimage
Flatpak can be kept in flathub repo since the updates are made by a bot

> I've outlined my reasons for leaving packaging in the repo. Nix absolutely requires to be in the same repo to my knowledge, and it's very convenient to be able to build the Deb straight in the repository. Especially for Jenkins, as it means a full rebuild is not necessary for the deb.

**Comment by muscaln** [comment]
> > Nix absolutely requires to be in the same repo to my knowledge
> 
> Hard to believe that every single program that nix supports has the packaging scripts in their repo. Why is it absolutely required specifically for this program?
> 


`nix` is not just a package manager. It also creates development environments from declared derivations. Setting up all required tools is straightforward(`nix develop`). We can move Nix derivation away from this repo but it will just make things more complex.

**Comment by ghost** [comment]
it's really sussy to build RPM packages in Arch so COPR will have to be done manually for now.

**Comment by d-513** [comment]
> it's really sussy to build RPM packages in Arch so COPR will have to be done manually for now.

No we can set up webhook rebuild
You can't upload prebuilt rpm to COPR anyway afaik

**Comment by d-513** [comment]
This PR turned to an issue about moving packaging rn lol also since we're using MPR I can close this until we do something like CI upload to MPR (and AUR in that case) or something

**Comment by ghost** [comment]
CI upload to AUR is finished.

I will however need to make an MPR account

**Comment by d-513** [comment]
> CI upload to AUR is finished.
> 
> 
> 
> I will however need to make an MPR account

Nice. When you do that send me your email or something so I can add you

How does the CI aur work? Update script I assume?

**Comment by ghost** [comment]
It `sed`s the pkgver, updpkgsums, makes the src info, commits, and pushes.

Anyways I registered now, swurl@swurl.xyz. @dada513

**Comment by d-513** [comment]
@binex-dsk ook I added u to all the packages

**Comment by ghost** [comment]
Jenkins now automatically pushes to both the AUR and MPR upon a package update.

The next step is automatically uploading new versions to COPR, @Jan200101 if you could provide details that would be cool and good

**Comment by Jan200101** [comment]
> Jenkins now automatically pushes to both the AUR and MPR upon a package update.
> 
> The next step is automatically uploading new versions to COPR, @Jan200101 if you could provide details that would be cool and good

Generally you would run the copr-cli utility (as available on [PyPi](https://pypi.org/project/copr-cli/)) and upload a new .src.rpm or .spec to it.

Generally copr should automatically fetch the needed Sources via the URL specified, though conflicts can come up between Distro releases and repo changes.

To upload stuff you need API access which can be gathered from here https://copr.fedorainfracloud.org/api/

**Comment by ghost** [comment]
Alright. Does COPR automatically build the RPM? If so then I can have Jenkins update the COPR package. I'd need to have access to push to sentry/polymc though

**Comment by Jan200101** [comment]
> Alright. Does COPR automatically build the RPM? If so then I can have Jenkins update the COPR package.

It does upon uploading either a .src.rpm or .spec

> I'd need to have access to push to sentry/polymc though

You would have to create an account either way and at that point it might be better to create a general polymc/polymc copr

**Comment by ghost** [comment]
Coronavirus disease (COVID-19)

image

**Comment by d-513** [comment]
Changed title and content of this PR.
Now it removes all debian packages from the repo, since they are maintained on the MPR.

**Comment by ghost** [comment]
I already removed the debian stuff but the gitignore I forgot

**Pull request merged**